src/app/task/create-task/create-task.component.ts

```diff
@@ -8,25 +8,39 @@ import { Router } from '@angular/router';
 import {MatInputModule} from '@angular/material/input';
 import { MatButtonModule } from '@angular/material/button';
 import { MatIconModule } from '@angular/material/icon';
-import { Subject, takeUntil } from 'rxjs';
+import { Subject, takeUntil, Observable } from 'rxjs';
+import { User } from '../../shared/models/user.model';
+import { UserService } from '../../shared/services/user.service';
+import {MatSelectModule} from '@angular/material/select';
+import { AsyncPipe } from '@angular/common';
 
 @Component({
   selector: 'create-task',
   standalone: true,
-  imports: [ReactiveFormsModule, TaskListComponent, MatInputModule, MatButtonModule, MatIconModule],
+  imports: [ReactiveFormsModule, AsyncPipe, MatSelectModule, TaskListComponent, MatInputModule, MatButtonModule, MatIconModule],
   templateUrl: './create-task.component.html',
   styleUrl: './create-task.component.scss'
 })
-export class CreateTaskComponent implements OnDestroy {
+export class CreateTaskComponent implements OnDestroy, OnInit {
   private destroy$: Subject<void> = new Subject<void>();
+  users$: Observable<User[]> | null = null;
 
   taskForm = new FormGroup({
     title: new FormControl('', Validators.required),
     description: new FormControl('', Validators.required),
     type: new FormControl('', Validators.required),
+    assignedTo: new FormControl('', Validators.required)
   });
 
-  constructor(private taskService: TaskService, private router: Router) {}
+  constructor(private taskService: TaskService, private userService: UserService, private router: Router) {}
+
+  ngOnInit(): void {
+    this.getUsers();
```

Comment:
you can call this.userService.getAllUsers() directly when defining the users$ observable

Reply:
since you only use that users$ observable to show the list so the user can choose who to assign the task to